### PR TITLE
Davem/mini

### PR DESCRIPTION
**Pull request by iabyn**
get minitest and Test -deparse to pass again

### CODE REVIEW

**Review by leonerd — APPROVED** [review]
LGTM

**Comment by jkeenan** [comment]
> get minitest and Test -deparse to pass again

At v5.39.7-175-ga1af65b9a8, I built an unthreaded perl on Linux, then called `make minitest`.  All tests passed.  So I don't see how you got two test failures needing correction.  @iabyn, can you post your `config_args` and any other data relevant to reproducing the failures?  Thanks.

File: pp_sys.c [hunk 1]
```diff
@@ -347,11 +347,13 @@ PP(pp_backtick)
  *
  * The first arg is the wildcard.
  *
- * The second arg is a gv which (for some reason) is just an empty
+ * The second arg is a gv which is just an empty
  * placeholder to temporarily assign to PL_last_in_gv. It's a GV unique to
  * this op with only a plain PVIO attached, which is in stash IO::File.
- * Presumably this is used because we tail-call do_readline(), which
- * expects PL_last_in_gv to be set.
+ * This because do_readline() expects PL_last_in_gv to be set by the
```
Comment (mauke) [inline pp_sys.c:353]: "This" -> "This is"?

File: pp_sys.c [hunk 2]
```diff
@@ -347,11 +347,13 @@ PP(pp_backtick)
...
+ * caller. In addition, when built PERL_EXTERNAL_GLOB (e.g. in miniperl),
```
Comment (mauke) [inline pp_sys.c:354]: "built" -> "built with"?

**Comment by iabyn** [comment]
On Sun, Feb 18, 2024 at 10:29:17AM -0800, James E Keenan wrote:
> > get minitest and Test -deparse to pass again
> 
> At v5.39.7-175-ga1af65b9a8, I built an unthreaded perl on Linux, then
> called `make minitest`.  All tests passed.  So I don't see how you got
> two test failures needing correction.  @iabyn, can you post your
> `config_args` and any other data relevant to reproducing the failures?

IIRC, it was threaded, with PERL_RC_STACK. The latter was probably the
important one.

-- 
Lear: Dost thou call me fool, boy?
Fool: All thy other titles thou hast given away; that thou wast born with.

**Comment by jkeenan** [comment]
> On Sun, Feb 18, 2024 at 10:29:17AM -0800, James E Keenan wrote: > get minitest and Test -deparse to pass again At v5.39.7-175-ga1af65b9a8, I built an unthreaded perl on Linux, then called `make minitest`. All tests passed. So I don't see how you got two test failures needing correction. @iabyn, can you post your `config_args` and any other data relevant to reproducing the failures?
> IIRC, it was threaded, with PERL_RC_STACK. The latter was probably the important one.
> […](#)

@iabyn, that doesn't appear to do the trick for me.  On Linux I built at v5.39.7-175-ga1af65b9a8   with this invocation:
```
$ sh ./Configure -des -Dusedevel -Duseithreads -Accflags='-DPERL_RC_STACK' && make test_prep
```
... which got me these `ccflags`:
```
$ ./perl -Ilib -V:ccflags
ccflags='-D_REENTRANT -D_GNU_SOURCE -DPERL_RC_STACK -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64';
```
But when I ran those tests:
```
$ cd t; TEST_JOBS=1 ./perl harness op/lex.t op/refstack.t; cd -
op/lex.t ....... ok     
op/refstack.t .. ok    
All tests successful.
Files=2, Tests=121,  1 wallclock secs ( 0.02 usr  0.00 sys +  0.16 cusr  0.07 csys =  0.25 CPU)
Result: PASS
```
So as yet I don't have something I can use to evaluate the impact of the patch.  Suggestions?

**Comment by jkeenan** [comment]
> > On Sun, Feb 18, 2024 at 10:29:17AM -0800, James E Keenan wrote: > get minitest and Test -deparse to pass again At v5.39.7-175-ga1af65b9a8, I built an unthreaded perl on Linux, then called `make minitest`. All tests passed. So I don't see how you got two test failures needing correction. @iabyn, can you post your `config_args` and any other data relevant to reproducing the failures?
> > IIRC, it was threaded, with PERL_RC_STACK. The latter was probably the important one.
> > […](#)
> 
> @iabyn, that doesn't appear to do the trick for me. On Linux I built at v5.39.7-175-ga1af65b9a8 with this invocation:
> 
> ```
> $ sh ./Configure -des -Dusedevel -Duseithreads -Accflags='-DPERL_RC_STACK' && make test_prep
> ```
> 
> ... which got me these `ccflags`:
> 
> ```
> $ ./perl -Ilib -V:ccflags
> ccflags='-D_REENTRANT -D_GNU_SOURCE -DPERL_RC_STACK -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64';
> ```
> 
> But when I ran those tests:
> 
> ```
> $ cd t; TEST_JOBS=1 ./perl harness op/lex.t op/refstack.t; cd -
> op/lex.t ....... ok     
> op/refstack.t .. ok    
> All tests successful.
> Files=2, Tests=121,  1 wallclock secs ( 0.02 usr  0.00 sys +  0.16 cusr  0.07 csys =  0.25 CPU)
> Result: PASS
> ```
> 
> So as yet I don't have something I can use to evaluate the impact of the patch. Suggestions?

Scratch that.  I see that the complaint was that these didn't PASS under `miniperl`.

**Comment by jkeenan** [comment]
I reconfigured with
```
$ sh ./Configure -des -Dusedevel -Duseithreads -Accflags='-DPERL_RC_STACK' && $ make minitest_prep
```
... which got me these `ccflags`:
```
$ ./miniperl -Ilib -V:ccflags
ccflags='-D_REENTRANT -D_GNU_SOURCE -DPERL_RC_STACK -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64';
```
But `miniperl -Ilib t/op/lex.t` PASSed for me.
```
$ ./miniperl -Ilib t/op/lex.t | ack '^not ok'
$ ./miniperl -Ilib t/op/lex.t | ack -c '^ok'
53
$ ./miniperl -Ilib t/op/refstack.t | ack -c '^not ok'
# Failed test 46 - GH \#15769 at t/op/refstack.t line 581
#      got "ERRNO hash is read only! at /home/jkeenan/gitwork/perl2/lib/Errno.pm line 207.\n\tErrno::STORE(Errno=HASH(0x55ec786e7bd0)) called at - line 1"
# expected ""
#  diff at 0
#    after ""
#     have "ERRNO hash is read only! at /home/jkeena"
#     want ""
# PROG: 
# map%$_= %_= %$_,%::;  map %$_ = %_, *::, $::{Internals::};
# STATUS: 65280
1
```
So, while I now see a rationale for a correction to get `t/op/refstack.t` to PASS under this configuration and `miniperl`, I still don't see that for `t/op/lex.t`.  Can you clarify?  Thanks.

**Comment by jkeenan** [comment]
The `GH \#15769` test in `t/op/refstack.t` has been failing under `miniperl` since the file was introduced into the core distro last August in a61d85e106.  Dave's branch simply skips that test under `miniperl`.  So that change is okay by me.

The comment which Dave added to `t/op/lex.t` suggests that the code change in that file is cautionary/preventive.  So that change is also okay by me.

Hence, assuming the changes in non-`t/op/*.t` files are okay with other reviewers, I have no objection to this merge.  Thanks.

**Comment by iabyn** [comment]
I've fixed the flagged typos and merged.

**Pull request closed without merging**